Permanent/Common/Base/SwiftUIViews/PullDownButton.swift

```diff
@@ -34,19 +34,33 @@ struct PullDownButton: View {
                                 .inset(by: 0.5)
                                 .stroke(Color(red: 0.96, green: 0.96, blue: 0.99), lineWidth: 1)
                         )
-                    HStack {
-                        Text(selectedItem?.title ?? "Select Item")
-                            .textStyle(SmallXXRegularTextStyle())
-                            .animation(.none)
-                        Spacer()
-                        Image(systemName: "chevron.down")
-                            .rotationEffect(.degrees( isSelecting ? -180 : 0))
-                        
+                    Button(action: {
+                        isSelecting = !isSelecting
```

Comment:
You have a button with the action isSelecting = !isSelecting. But you also added a tap gesture to the button to toggle the is selecting. Could you explain the logic behind this?